assets/src/block-editor/QueryBlockExtension/QueryBlockExtension.js

```diff
@@ -0,0 +1,126 @@
+import {RadioControl, PanelBody} from '@wordpress/components';
+import {InspectorControls} from '@wordpress/block-editor';
+import {useCallback, useEffect, useMemo, useState} from '@wordpress/element';
+
+const {addFilter} = wp.hooks;
+const {__} = wp.i18n;
+
+const targetBlocks = ['core/post-template'];
+
+const layoutTypes = [
+  {label: 'Grid', value: 'grid', columnCount: 3},
+  {label: 'Carousel', value: 'flex', columnCount: 6},
+  // {label: 'List', value: 'default', columnCount: 3}, // This might be enabled within Posts List
+];
+
+export const setupQueryBlockExtension = function() {
+  const {createHigherOrderComponent} = wp.compose;
+
+  addFilter(
+    'editor.BlockEdit',
+    'planet4-blocks/overrides/query-loop-layout',
+    createHigherOrderComponent(
+      BlockEdit => props => {
+        if (props.attributes.namespace === 'planet4-blocks/actions-list' || targetBlocks.includes(props.name)) {
```

Comment:
Shouldn't it be `&&` instead of `||`? Won't this apply the change to all `core/post-template` blocks?

Reply:
This basically because the essence of blocks is to work isolated by other blocks. you can get the `clientId` of its parent and update its attributes but it's looks like as an anti-pattern.

So, `props.attributes.namespace === 'planet4-blocks/actions-list'` is required to render the new layout view options and `targetBlocks.includes(props.name)` is checked to hide the layout options from toolbar and avoid to get confusions with editors.

Does it make sense?